sktime/forecasting/compose/_pipeline.py

```diff
@@ -1448,3 +1450,192 @@ def get_test_params(cls, parameter_set="default"):
         params = {"forecaster_X": fx, "forecaster_y": fy}
 
         return params
+
+
+class Permute(_DelegatedForecaster, _HeterogenousMetaEstimator):
+    """Permutation compositor for permuting pipeline steps.
+
+    The compositor can be used to permute the sequence of any meta-estimator,
+    including ForecastingPipeline and TransformedTargetForecaster.
+
+    The `steps_arg` parameter needs to be pointed to the "steps"-like parameter
+    of the wrapped forecaster and `permutation` switches the sequence of steps.
+
+    Not very useful on its own, but
+    useful in combination with tuning or auto-ML wrappers on top of this.
+
+    Parameters
+    ----------
+    estimator : sktime forecaster, inheriting from BaseForecaster
+        must have parameter with name `steps_arg`
+        estimator whose steps are being permuted
+    permutation : list of str, or None, optional, default = None
+        if not None, must be equal length as getattr(estimator, steps_arg)
+        and elements must be equal to names of estimator.steps_arg estimators
+        names are unique names as created by _get_estimator_tuples (if unnamed list),
+        or first string element of tuples, of estimator.steps_arg
+        list is interpreted as range of permutation of names
+        if None, is interpreted as the identity permutation
+    steps_arg : string, optional, default="steps"
+        name of the steps parameter. getattr(estimator, steps_arg) must be
+        list of estimators, or list of (str, estimator) pairs
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.base import ForecastingHorizon
+    >>> from sktime.forecasting.compose import ForecastingPipeline, Permute
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.transformations.series.boxcox import BoxCoxTransformer
+    >>> from sktime.transformations.series.exponent import ExponentTransformer
+
+    Simple example: permute sequence of estimator in forecasting pipeline
+    >>> y = load_airline()
+    >>> fh = ForecastingHorizon([1, 2, 3])
+    >>> pipe = ForecastingPipeline(
+    ...     [
+    ...         ("boxcox", BoxCoxTransformer()),
+    ...         ("exp", ExponentTransformer(3)),
+    ...         ("naive", NaiveForecaster()),
+    ...     ]
+    ... )
+    >>> # this results in the pipeline with sequence "exp", "boxcox", "naive"
+    >>> permuted = Permute(pipe, ["exp", "boxcox", "naive"])
```

Comment:
```suggestion
    >>> permuted = Permute(pipe, ["exp", "boxcox", "naive"])
    >>> permuted = permuted.fit(y, fh-fh)
    >>> y_pred = permuted.predict()
```
In order for it to match up with my understanding of what this estimator is doing I think it should be like this ^

Reply:
confused, what you wrote seems to be identical with what I wrote?

Reply:
I thought I had highlighted multiple lines for suggested change.  Basically - the line after this should be permuted.fit()  - yes?

Reply:
oh, of course

Reply:
fixed